python/python/lance/vector.py

```diff
@@ -0,0 +1,84 @@
+"""Embedding vector utilities"""
+from typing import Union, Optional
+
+import numpy as np
+import pyarrow as pa
+
+#  Copyright 2023 Lance Developers
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+def _normalize_vectors(vectors, ndim):
+    if ndim is None:
+        ndim = len(next(iter(vectors)))
+    values = np.array(vectors, dtype="float32").ravel()
+    return pa.FixedSizeListArray.from_arrays(values, list_size=ndim)
+
+
+def _validate_ndim(values, ndim):
+    for v in values:
+        if ndim is None:
+            ndim = len(v)
+        else:
+            if ndim != len(v):
+                raise ValueError(f"Expected {ndim} dimensions but got {len(v)} for {v}")
+    return ndim
+
+
+def to_vectors(data: Union[dict, list, np.ndarray],
+               names: Optional[Union[str, list[str]]] = None,
+               ndim: Optional[int] = None,
+               check_ndim: bool = True) -> pa.Table:
+    """
+    Create a pyarrow Table containing vectors.
+    Vectors are created as FixedSizeListArray's in pyarrow with Float32 values.
+
```

Comment:
Maybe add an example section here? The usage is not clear.

Reply:
added